pkg/storage/util/util.go

```diff
@@ -83,7 +79,7 @@ func CreateOrUpdateSecret(name string, namespace string, data map[string]string)
 	var updatedSecret *coreapi.Secret
 
 	if err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
-		cur, err := client.Secrets(namespace).Get(name, metaapi.GetOptions{})
+		cur, err := listers.Secrets.Get(name)
```

Comment:
same here about going to the source vs the cache since we're trying to ensure we have the latest content

Reply:
The retry is technically for the create/update that happens towards the end of the function, not the Get, as I don't believe that would return a conflict type error.

Reply:
Updated per our conversation